### ISSUE TITLE
`Request-EsxiPasswordExpiration` error handling fails

### ISSUE THREAD

**Issue by nzRegularIT** [vmware/powershell-module-for-vmware-cloud-foundation-password-management#61]
### Code of Conduct

- [X] I have read and agree to the [Code of Conduct](https://github.com/vmware/powershell-module-for-vmware-cloud-foundation-password-management/blob/main/CODE_OF_CONDUCT.md).
- [ ] Vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue initial description to help the maintainers prioritize.
- [ ] Do not leave "+1" or other comments that do not add relevant information or questions.
- [X] If you are interested in working on this issue or have submitted a pull request, please leave a comment.

### VMware Cloud Foundation

v4.5

### PowerShell Version

PowerShell Core 7.2.8

### PowerCLI Version

13.1.0.21624340

### Module Version

1.1.0.1008

### PowerValidatedSolutions Version

2.3.0.1007

### PowerVCF Version

2.3.0.1004

### Guest Operating System

Windows Server 2019

### Description

`Request-EsxiPasswordExpiration` returns an error upon execution in PowerShell Core

### Error or Debug Output

[06-27-2023_15:41:56]   Error at Script Line 6103
Write-LogMessage: Cannot bind argument to parameter 'Path' because it is null.

### Expected Behavior

Output should display ESXi Password Expiration details for the given workload domain and cluster

### Actual Behavior

`Debug-CatchWriter` appears to throw an error due when running the `Write-LogMessage` function in `PowerVCF`. This appears to be due to the line `Add-Content -Path $logFile $logContent` in function `Write-LogMessage`.

```powershell
Function Debug-CatchWriter {
    Param (
        [Parameter (Mandatory = $true)] [PSObject]$object
    )

    $lineNumber = $object.InvocationInfo.ScriptLineNumber
    $lineText = $object.InvocationInfo.Line.trim()
    $errorMessage = $object.Exception.Message
    Write-LogMessage -message " Error at Script Line $lineNumber" -colour Red
    Write-LogMessage -message " Relevant Command: $lineText" -colour Red
    Write-LogMessage -message " Error Message: $errorMessage" -colour Red
}
Export-ModuleMember -Function Debug-CatchWriter

Function Write-LogMessage {
    Param (
        [Parameter (Mandatory = $true)] [AllowEmptyString()] [String]$Message,
        [Parameter (Mandatory = $false)] [ValidateSet("INFO", "ERROR", "WARNING", "EXCEPTION")] [String]$type,
        [Parameter (Mandatory = $false)] [String]$Colour,
        [Parameter (Mandatory = $false)] [String]$Skipnewline
    )

    if (!$Colour) {
        $Colour = "White"
    }

    $timeStamp = Get-Date -Format "MM-dd-yyyy_HH:mm:ss"

    Write-Host -NoNewline -ForegroundColor White " [$timestamp]"
    if ($Skipnewline) {
        Write-Host -NoNewline -ForegroundColor $Colour " $type $Message"        
    }
    else {
        Write-Host -ForegroundColor $colour " $Type $Message" 
    }
    $logContent = '[' + $timeStamp + '] ' + $Type + ' ' + $Message
    Add-Content -Path $logFile $logContent
}
Export-ModuleMember -Function Write-LogMessage
```

### Steps to Reproduce

```powershell
Set-PSRepository -Name PSGallery -InstallationPolicy Trusted
Install-Module -Name VMware.PowerCLI -MinimumVersion 13.0.0
Install-Module -Name VMware.vSphere.SsoAdmin -MinimumVersion 1.3.9
Install-Module -Name PowerVCF -MinimumVersion 2.3.0
Install-Module -Name PowerValidatedSolutions -MinimumVersion 2.3.0
Install-Module -Name VMware.CloudFoundation.PasswordManagement
```

Open PowerShell Core 7.2.8

```powershell
Request-EsxiPasswordExpiration -server $sddcManagerFQDN -user $sddcManagerAdminUser -pass $sddcManagerAdminPassword -domain $sddcManagerDomain -cluster $sddcManagerCluster
```

### References

I found two workarounds while testing:

1. Update `Debug-CatchWriter` to `Debug-ExceptionWriter`
2. Comment out the 3 `Write-LogMessage` lines in the `Debug-CatchWriter` function in PowerVCF

```powershell
Function Debug-CatchWriter {
    Param (
        [Parameter (Mandatory = $true)] [PSObject]$object
    )

    $lineNumber = $object.InvocationInfo.ScriptLineNumber
    $lineText = $object.InvocationInfo.Line.trim()
    $errorMessage = $object.Exception.Message
    #Write-LogMessage -message " Error at Script Line $lineNumber" -colour Red
    #Write-LogMessage -message " Relevant Command: $lineText" -colour Red
    #Write-LogMessage -message " Error Message: $errorMessage" -colour Red
}
Export-ModuleMember -Function Debug-CatchWriter
```

**Comment by tenthirtyam** [comment]
Could you confirm the module and dependencies are imported?

**Comment by sowjuec** [comment]
After installation of required powershell modules, for power-shell core, importing of the module needs to be done. Also you can check `Test-VcfPasswordManagementPrereq`, to see if all required versions of the module are installed.
[UserGuide For Reference](https://vmware.github.io/powershell-module-for-vmware-cloud-foundation-password-management/install/)

Could you please attach the output of `Test-VcfPasswordManagementPrereq`command?

Once I followed the steps, I am not seeing any issues. So I am unable to reproduce this.
```powershell
PS C:\GitHubPublic\VMware.CloudFoundation.PowerManagement2\docs> Set-PSRepository -Name PSGallery -InstallationPolicy Trusted
>> Install-Module -Name VMware.PowerCLI -MinimumVersion 13.0.0
>> Install-Module -Name VMware.vSphere.SsoAdmin -MinimumVersion 1.3.9
>> Install-Module -Name PowerVCF -MinimumVersion 2.3.0 -repository 'PSGallery' -force
>> Install-Module -Name PowerValidatedSolutions -MinimumVersion 2.3.0 -repository 'PSGallery' -force
>> Install-Module -Name VMware.CloudFoundation.PasswordManagement -repository 'PSGallery'-force
PS C:\GitHubPublic\VMware.CloudFoundation.PowerManagement2\docs> get-module

ModuleType Version    PreRelease Name                                ExportedCommands
---------- -------    ---------- ----                                ----------------
Binary     7.0.0.0               CimCmdlets                          {Get-CimAssociatedInstance, Get-CimClass, Get-CimInstance, Get-CimSession…}
Manifest   1.0.0.0               DnsClient                           {Resolve-DnsName, Add-DnsClientNrptRule, Clear-DnsClientCache, Get-DnsClient…}
Manifest   7.0.0.0               Microsoft.PowerShell.Management     {Add-Content, Clear-Content, Clear-Item, Clear-ItemProperty…}
Manifest   7.0.0.0               Microsoft.PowerShell.Utility        {Add-Member, Add-Type, Clear-Variable, Compare-Object…}
Manifest   2.0.0.0               NetAdapter                          {Disable-NetAdapter, Disable-NetAdapterBinding, Disable-NetAdapterChecksumOffload, Disable-NetAdapterEncapsulatedPacketTaskOffload…}
Manifest   2.0.0.0               NetSecurity                         {Get-DAPolicyChange, New-NetIPsecAuthProposal, New-NetIPsecMainModeCryptoProposal, New-NetIPsecQuickModeCryptoProposal…}
Manifest   1.0.0.0               NetTCPIP                            {Find-NetRoute, Get-NetCompartment, Get-NetIPAddress, Get-NetIPConfiguration…}
Script     1.4.8.1               PackageManagement                   {Find-Package, Find-PackageProvider, Get-Package, Get-PackageProvider…}
Script     0.14.2                platyPS                             {Get-HelpPreview, Get-MarkdownMetadata, Merge-MarkdownHelp, New-ExternalHelp…}
Manifest   3.0.8                 Posh-SSH                            {Get-SCPItem, Get-SFTPItem, Get-SSHHostKey, Get-SSHJsonKnownHost…}
Script     2.2.5                 PowerShellGet                       {Find-Command, Find-DscResource, Find-Module, Find-RoleCapability…}
Script     2.3.0.1004            PowerVCF                            {Add-VCFApplicationVirtualNetwork, Add-VCFNetworkIPPool, Connect-CloudBuilder, Debug-CatchWriter…}
Script     2.2.6                 PSReadLine                          {Get-PSReadLineKeyHandler, Get-PSReadLineOption, Remove-PSReadLineKeyHandler, Set-PSReadLineKeyHandler…}
Script     1.1.0.1001            VMware.CloudFoundation.PowerManage… {Get-DrsAutomationLevel, Get-EdgeNodeFromNSXManager, Get-MaintenanceMode, Get-NSXTComputeManagers…}
Script     13.0.0.20…            VMware.VimAutomation.Core           {Add-PassthroughDevice, Add-VirtualSwitchPhysicalNetworkAdapter, Add-VMHost, Add-VMHostNtpServer…}
Script     13.0.0.20…            VMware.VimAutomation.Storage        {Add-EntityDefaultKeyProvider, Add-KeyManagementServer, Add-VsanFileServiceOvf, Add-VsanObjectToRepairQueue…}

PS C:\GitHubPublic\VMware.CloudFoundation.PowerManagement2\docs> get-installedModule

Version              Name                                Repository           Description
-------              ----                                ----------           -----------
3.0.8                Posh-SSH                            PSGallery            Provide SSH and SCP functionality for executing commands against remote hosts.
2.4.0.1008           PowerValidatedSolutions             PSGallery            PowerShell Module for VMware Validated Solutions
2.3.0.1004           PowerVCF                            PSGallery            PowerShell Module for the VMware Cloud Foundation 4.x API
1.2.0.1005           VMware.CloudFoundation.PasswordMan… PSGallery            PowerShell Module for VMware Cloud Foundation Password Management

PS C:\GitHubPublic\VMware.CloudFoundation.PowerManagement2\docs> Import-Module -Name VMware.PowerCLI
>> Import-Module -Name VMware.vSphere.SsoAdmin
>> Import-Module -Name PowerVCF
>> Import-Module -Name PowerValidatedSolutions
>> Import-Module -Name VMware.CloudFoundation.PasswordManagement
          Welcome to VMware PowerCLI!

Log in to a vCenter Server or ESX host:              Connect-VIServer
To find out what commands are available, type:       Get-VICommand
To show searchable help for all PowerCLI commands:   Get-PowerCLIHelp
Once you've connected, display all virtual machines: Get-VM
If you need more help, visit the PowerCLI community: Get-PowerCLICommunity

       Copyright (C) VMware, Inc. All rights reserved.
```

**Comment by nzRegularIT** [comment]
Great thanks @sowjuec @tenthirtyam! I think this bug may be related to two others that I have filed, and this one may disappear once the other two are confirmed/resolved? What do you think?

I have retested and receive the following error, which I believe is related to https://github.com/vmware/powershell-module-for-vmware-cloud-foundation-password-management/issues/59

```
PS C:\Windows\System32> Request-EsxiPasswordExpiration

cmdlet Request-EsxiPasswordExpiration at command pipeline position 1
Supply values for the following parameters:
server: <redacted>
user: <redacted>
pass: <redacted>
domain: <redacted>
cluster: <redacted>
 Error at Script Line 6386
 Relevant Command: if (Test-Connection -server $server) {
 Error Message: A parameter cannot be found that matches parameter name 'server'.
```

For Test-VcfPasswordManagementPrereq, I receive the following error, which is related to https://github.com/vmware/powershell-module-for-vmware-cloud-foundation-password-management/issues/58

```
 [06-28-2023_16:50:34] [ERROR] PowerShell Module: VMware.PowerCLI 13.0.0 minimum required version is not installed.
```

### Module Summary

```
Import-Module -Name VMware.PowerCLI
Import-Module -Name VMware.vSphere.SsoAdmin
Import-Module -Name PowerVCF
Import-Module -Name PowerValidatedSolutions
Import-Module -Name VMware.CloudFoundation.PasswordManagement
```

```
PS C:\Windows\System32> Get-Module

ModuleType Version    PreRelease Name                                ExportedCommands
---------- -------    ---------- ----                                ----------------
Manifest   7.0.0.0               Microsoft.PowerShell.Management     {Add-Content, Clear-Content, Clear-Item, Clear-ItemProperty…}
Manifest   7.0.0.0               Microsoft.PowerShell.Utility        {Add-Member, Add-Type, Clear-Variable, Compare-Object…}
Script     1.4.7                 PackageManagement                   {Find-Package, Find-PackageProvider, Get-Package, Get-PackageProvider…}
Script     2.2.5                 PowerShellGet                       {Find-Command, Find-DscResource, Find-Module, Find-RoleCapability…}
Script     2.3.0.1007            PowerValidatedSolutions             {Add-AntiAffinityRule, Add-CEPTrustedCertificate, Add-CEPvCenterServer, Ad…
Script     2.3.0.1004            PowerVCF                            {Add-VCFApplicationVirtualNetwork, Add-VCFNetworkIPPool, Connect-CloudBuil…
Script     2.1.0                 PSReadLine                          {Get-PSReadLineKeyHandler, Get-PSReadLineOption, Remove-PSReadLineKeyHandl…
Script     1.2.0.1005            VMware.CloudFoundation.PasswordMan… {Get-PasswordPolicyConfig, Get-PasswordPolicyDefault, Invoke-PasswordPolic…
Script     12.6.0.19…            VMware.CloudServices                {Connect-Vcs, Disconnect-Vcs, Get-VcsOrganizationRole, Get-VcsService…}
Script     8.0.0.216…            VMware.DeployAutomation             {Add-CustomCertificate, Add-DeployRule, Add-ProxyServer, Add-ScriptBundle…}
Script     8.0.0.216…            VMware.ImageBuilder                 {Add-EsxSoftwareDepot, Add-EsxSoftwarePackage, Compare-EsxImageProfile, Ex…
Manifest   13.1.0.21…            VMware.PowerCLI
Script     12.6.0.19…            VMware.PowerCLI.Sdk
Script     12.6.0.19…            VMware.PowerCLI.Sdk.Types
Script     12.6.0.19…            VMware.PowerCLI.VCenter             {Add-VITrustedCertificate, Get-VIApplianceService, Get-VIMachineCertificat…
Script     12.6.0.19…            VMware.PowerCLI.VCenter.Types.Appl…
Script     12.6.0.19…            VMware.PowerCLI.VCenter.Types.Cert…
Script     4.1.0.216…            VMware.Sdk.Nsx.Policy               {Connect-NsxServer, Connect-NsxVmcServer, Disconnect-NsxServer, Disconnect…
Script     1.0.1111.…            VMware.Sdk.Runtime                  {Add-ServerConfiguration, ConvertFrom-JsonX, Get-ServerConfiguration, Invo…
Script     8.7.0.216…            VMware.Sdk.Srm                      {Connect-SrmSdkServer, Disconnect-SrmSdkServer, Get-SrmOperation, Initiali…
Script     8.7.0.216…            VMware.Sdk.Vr                       {Connect-VrServer, Disconnect-VrServer, Get-VrOperation, Initialize-VrConf…
Manifest   8.0.1111.…            VMware.Sdk.vSphere
Script     8.0.1111.…            VMware.Sdk.vSphere.Appliance        {Initialize-HealthCheckSettingsUpdateSpec, Initialize-LocalAccountsConfig,…
Script     8.0.1111.…            VMware.Sdk.vSphere.Appliance.Access {Initialize-AccessConsolecliSetRequestBody, Initialize-AccessDcuiSetReques…
Script     8.0.1111.…            VMware.Sdk.vSphere.Appliance.Health {Invoke-GetHealthApplmgmt, Invoke-GetHealthDatabase, Invoke-GetHealthDatab…
Script     8.0.1111.…            VMware.Sdk.vSphere.Appliance.Infra… {Initialize-InfraprofileConfigsImportProfileSpec, Initialize-InfraprofileC…
Script     8.0.1111.…            VMware.Sdk.vSphere.Appliance.Local… {Initialize-LocalAccountsPolicyInfo, Invoke-GetLocalAccountsGlobalPolicy, …
Script     8.0.1111.…            VMware.Sdk.vSphere.Appliance.Loggi… {Initialize-LoggingForwardingConfig, Initialize-LoggingForwardingSetReques…
Script     8.0.1111.…            VMware.Sdk.vSphere.Appliance.Netwo… {Initialize-NetworkingDnsDomainsAddRequestBody, Initialize-NetworkingDnsDo…
Script     8.0.1111.…            VMware.Sdk.vSphere.Appliance.Recov… {Initialize-RecoveryBackupBackupRequest, Initialize-RecoveryBackupJobBacku…
Script     8.0.1111.…            VMware.Sdk.vSphere.Appliance.Suppo… Invoke-GetSupportBundleComponents
Script     8.0.1111.…            VMware.Sdk.vSphere.Appliance.System {Initialize-SystemSecurityGlobalFipsUpdateSpec, Initialize-SystemTimeTimez…
Script     8.0.1111.…            VMware.Sdk.vSphere.Appliance.Update {Initialize-UpdatePendingInstallRequestBody, Initialize-UpdatePendingStage…
Script     8.0.1111.…            VMware.Sdk.vSphere.Cis              {Initialize-TasksFilterSpec, Initialize-TasksGetSpec, Initialize-TasksList…
Script     8.0.1111.…            VMware.Sdk.vSphere.Cis.Tagging      {Initialize-StdDynamicID, Initialize-TaggingCategoryAddToUsedByRequestBody…
Script     8.0.1111.…            VMware.Sdk.vSphere.Content          {Initialize-ConfigurationModel, Initialize-LibraryFindSpec, Initialize-Lib…
Script     8.0.1111.…            VMware.Sdk.vSphere.ContentLibrary   {Initialize-LibraryItemCertificateVerificationInfo, Initialize-LibraryItem…
Script     8.0.1111.…            VMware.Sdk.vSphere.Esx.Hcl          {Initialize-HclHostsCompatibilityReportSpec, Invoke-CreateHostCompatibilit…
Script     8.0.1111.…            VMware.Sdk.vSphere.Esx.Hosts        {Initialize-HostsSoftwareHostCredentials, Invoke-GetSoftware, Invoke-ListH…
Script     8.0.1111.…            VMware.Sdk.vSphere.Esx.Settings     {Initialize-SettingsAddOnSpec, Initialize-SettingsBaseImageSpec, Initializ…
Script     8.0.1111.…            VMware.Sdk.vSphere.VAPI.Metadata    {Initialize-MetadataCliCommandGetRequestBody, Initialize-MetadataCliComman…
Script     8.0.1111.…            VMware.Sdk.vSphere.vCenter          {Initialize-DatacenterCreateSpec, Initialize-HostCreateSpec, Initialize-Re…
Script     8.0.1111.…            VMware.Sdk.vSphere.vCenter.Authent… Invoke-CreateAuthenticationToken
Script     8.0.1111.…            VMware.Sdk.vSphere.vCenter.Authori… {Initialize-AuthorizationPrivilegeChecksFilterSpec, Initialize-Authorizati…
Script     8.0.1111.…            VMware.Sdk.vSphere.vCenter.CertMan… {Initialize-CertificateManagementVcenterSigningCertificateRefreshRequestBo…
Script     8.0.1111.…            VMware.Sdk.vSphere.vCenter.Consump… {Initialize-ConsumptionDomainsZonesCreateSpec, Invoke-AddZoneAssociations,…
Script     8.0.1111.…            VMware.Sdk.vSphere.vCenter.Content  {Initialize-ContentRegistriesHarborCreateSpec, Initialize-ContentRegistrie…
Script     8.0.1111.…            VMware.Sdk.vSphere.vCenter.Datasto… Invoke-GetDatastoreDefaultPolicy
Script     8.0.1111.…            VMware.Sdk.vSphere.vCenter.Deploym… {Initialize-DeploymentHistoryMigrationSpec, Initialize-DeploymentImportHis…
Script     8.0.1111.…            VMware.Sdk.vSphere.vCenter.Guest    {Initialize-GuestCustomizationSpecsCreateSpec, Initialize-GuestCustomizati…
Script     8.0.1111.…            VMware.Sdk.vSphere.vCenter.hvc      {Initialize-HvcLinksCreateSpec, Initialize-HvcLinksCredentials, Initialize…
Script     8.0.1111.…            VMware.Sdk.vSphere.vCenter.Identity {Initialize-IdentityProvidersActiveDirectoryOverLdap, Initialize-IdentityP…
Script     8.0.1111.…            VMware.Sdk.vSphere.vCenter.Invento… {Invoke-GetInventoryDatastore, Invoke-GetInventoryNetwork}
Script     8.0.1111.…            VMware.Sdk.vSphere.vCenter.ISO      {Initialize-IsoImageMountRequestBody, Initialize-IsoImageUnmountRequestBod…
Script     8.0.1111.…            VMware.Sdk.vSphere.vCenter.LCM      {Initialize-LcmDiscoveryAssociatedProductsCreateSpec, Initialize-LcmDiscov…
Script     8.0.1111.…            VMware.Sdk.vSphere.vCenter.Namespa… {Initialize-NamespaceManagementClustersEnableSpec, Initialize-NamespaceMan…
Script     8.0.1111.…            VMware.Sdk.vSphere.vCenter.Namespa… {Initialize-NamespacesAccessCreateSpec, Initialize-NamespacesAccessSetSpec…
Script     8.0.1111.…            VMware.Sdk.vSphere.vCenter.OVF      {Initialize-OvfLibraryItemCreateRequestBody, Initialize-OvfLibraryItemCrea…
Script     8.0.1111.…            VMware.Sdk.vSphere.vCenter.Services {Initialize-ServicesServiceUpdateSpec, Invoke-GetService, Invoke-GetServic…
Script     8.0.1111.…            VMware.Sdk.vSphere.vCenter.Storage  {Initialize-StoragePoliciesCheckCompatibilityRequestBody, Invoke-CheckComp…
Script     8.0.1111.…            VMware.Sdk.vSphere.vCenter.SystemC… {Initialize-DeploymentRemotePscSpec, Initialize-SystemConfigDeploymentType…
Script     8.0.1111.…            VMware.Sdk.vSphere.vCenter.Tagging  Invoke-ListAssociations
Script     8.0.1111.…            VMware.Sdk.vSphere.vCenter.Topology {Invoke-GetNodeTopology, Invoke-ListTopologyNodes, Invoke-ListTopologyRepl…
Script     8.0.1111.…            VMware.Sdk.vSphere.vCenter.Trusted… {Initialize-TrustedInfrastructureAttestationServicesCreateSpec, Initialize…
Script     8.0.1111.…            VMware.Sdk.vSphere.vCenter.VCHA     {Initialize-VchaClusterActiveGetRequestBody, Initialize-VchaClusterActiveS…
Script     8.0.1111.…            VMware.Sdk.vSphere.vCenter.Vm       {Initialize-GuestAdapterMapping, Initialize-GuestCloudConfiguration, Initi…
Script     8.0.1111.…            VMware.Sdk.vSphere.vCenter.VmTempl… {Initialize-VmTemplateLibraryItemsCheckOutsCheckInSpec, Initialize-VmTempl…
Script     8.0.1111.…            VMware.Sdk.vSphere.vStats           {Initialize-AcqSpecsCounterSpec, Initialize-AcqSpecsCreateSpec, Initialize…
Script     8.0.1111.…            VMware.Sdk.vSphereRuntime           {Convert-InputStructure, Convert-OutputBody, Format-Body, Format-Headers…}
Script     8.1.0.216…            VMware.Vim
Script     13.1.0.21…            VMware.VimAutomation.Cis.Core       {Connect-CisServer, Disconnect-CisServer, Get-CisService}
Script     13.1.0.21…            VMware.VimAutomation.Cloud          {Add-CIDatastore, Connect-CIServer, Disconnect-CIServer, Get-Catalog…}
Script     13.1.0.21…            VMware.VimAutomation.Common         {Get-PowerCLIContext, Get-Task, New-OAuthSecurityContext, Stop-Task…}
Script     13.1.0.21…            VMware.VimAutomation.Core           {Add-PassthroughDevice, Add-VirtualSwitchPhysicalNetworkAdapter, Add-VMHos…
Script     13.0.0.20…            VMware.VimAutomation.Hcx            {Connect-HCXServer, Disconnect-HCXServer, Get-HCXAppliance, Get-HCXCompute…
Script     13.1.0.21…            VMware.VimAutomation.HorizonView    {Connect-HVServer, Disconnect-HVServer}
Script     12.0.0.15…            VMware.VimAutomation.License        Get-LicenseDataManager
Script     13.1.0.21…            VMware.VimAutomation.Nsxt           {Connect-NsxtServer, Disconnect-NsxtServer, Get-NsxtGlobalManagerService, …
Script     13.1.0.21…            VMware.VimAutomation.Sdk            {Get-ErrorReport, EnableParameterCompleters, Get-InstallPath, Get-PSVersio…
Script     13.1.0.21…            VMware.VimAutomation.Security       {Add-AttestationServiceInfo, Add-KeyProviderServiceInfo, Add-TrustAuthorit…
Script     12.7.0.20…            VMware.VimAutomation.Srm            {Connect-SrmServer, Disconnect-SrmServer}
Script     13.1.0.21…            VMware.VimAutomation.Storage        {Add-EntityDefaultKeyProvider, Add-KeyManagementServer, Add-VsanFileServic…
Script     1.6.0.0               VMware.VimAutomation.StorageUtility Update-VmfsDatastore
Script     13.1.0.21…            VMware.VimAutomation.Vds            {Add-VDSwitchPhysicalNetworkAdapter, Add-VDSwitchVMHost, Export-VDPortGrou…
Script     13.0.0.20…            VMware.VimAutomation.Vmc            {Add-VmcSddcHost, Connect-Vmc, Disconnect-Vmc, Get-AwsAccount…}
Script     13.1.0.21…            VMware.VimAutomation.vROps          {Connect-OMServer, Disconnect-OMServer, Get-OMAlert, Get-OMAlertDefinition…
Script     12.4.0.18…            VMware.VimAutomation.WorkloadManag… {Disable-WMCluster, Enable-WMCluster, Get-WMCluster, Get-WMNamespace…}
Script     1.3.9                 VMware.vSphere.SsoAdmin             {Add-GroupToSsoGroup, Add-LDAPIdentitySource, Add-UserToSsoGroup, Connect-…
Script     12.7.0.20…            VMware.VumAutomation                {Add-EntityBaseline, Copy-Patch, Get-Baseline, Get-Compliance…}
```

**Comment by tenthirtyam** [comment]
re: https://github.com/vmware/powershell-module-for-vmware-cloud-foundation-password-management/issues/61#issuecomment-1610727209

Yes, this should be resolved by #59 and the related pull request in the `PowerValidatedSolutions` dependency.

Once those updates are made in the dependency this should be resolved.

Ryan

**Comment by tenthirtyam** [comment]
Should also be addressed with the help of https://github.com/vmware/powershell-module-for-vmware-cloud-foundation-password-management/pull/63.
